README.md

```diff
@@ -12,15 +12,16 @@ Setup an ALB with related resources.
 
 ### Available variables
 
-* [`name_prefix`]: String(required): Name prefix of the ALB and security group
+* [`name_prefix`]: String(optional): Name prefix of the ALB and security group
```

Comment:
To keep full backwards compatibility and make it more understandable this is how it should work imo:

- set default of `name_prefix` to an empty string
- set default of `name` as `null`

- if none are provided, the name will be completely randomised
- if only `name_prefix` is provided it'll keep the current behaviour
- setting the `name` variable overrides whatever is set in `name_prefix` and sets the "${var.project}-${var.environment}-${var.name}" name style